src/App.js

```diff
@@ -259,13 +283,43 @@ class App extends React.Component {
     }, 1000)
   }
 
+  handleIdentityCancel = () => {
+    this.setState({ identityIntent: null })
+  }
+
+  handleIdentitySave = ({ address, label }) => {
+    const { resolve, reject } = this.state.identityIntent
+    this.state.wrapper
+      .modifyAddressIdentity(address, { name: label })
+      .then(() => this.setState({ identityIntent: null }, resolve))
+      .catch(reject)
+  }
+
+  handleIdentityResolve = address => {
```

Comment:
Can you clarify how and when this function rejects the promise

Reply:
This function does not really resolve/reject, it does return the `resolveAddressIdentity` function from `this.state.wrapper` that resolves when the `identity` has been modified or rejects if the user cancels/clicks-outside in the modal interaction.